### PR TITLE
Added support for subfolders in assets, and named object headers & models.

### PR DESCRIPTION
**Pull request by DavidSM64**
Renaming individual files (like textures) should be a lot easier now.

### CODE REVIEW

File: include/asset_enums.h [hunk 1]
```diff
@@ -2654,9 +2654,9 @@ typedef enum {
 
 typedef enum {
     ASSET_MENU_TEXT_ENGLISH,
-    ASSET_MENU_TEXT_FRENCH,
-    ASSET_MENU_TEXT_GERMAN,
-    ASSET_MENU_TEXT_JAPANESE
+    ASSET_MENU_TEXT_1,
```
Comment (Ryan-Myers) [inline include/asset_enums.h:2657]: Do we want these to lose their names?

**Review by Ryan-Myers — APPROVED** [review]
LGTM, but can we get the language asset names back?

File: include/asset_enums.h (hunk 1 above)
Comment (DavidSM64) [reply include/asset_enums.h:2657]: I didn't see that! I'll look into it.

Comment (DavidSM64) [reply include/asset_enums.h]: Fixed!

**Pull request merged**